tests/tests/swfs/avm2/matrix/Test.as

```diff
@@ -0,0 +1,410 @@
+trace("new Matrix(1, 2)", new Matrix(1, 2));
+trace("new Matrix(1, 2, 3)", new Matrix(1, 2, 3));
+trace("new Matrix(1, 2, 3, 4)", new Matrix(1, 2, 3, 4));
+trace("new Matrix(1, 2, 3, 4, 5)", new Matrix(1, 2, 3, 4, 5));
+trace("new Matrix(1, 2, 3, 4, 5, 6)", new Matrix(1, 2, 3, 4, 5, 6));
+
+// cannot use NaN values with approximate tester
+// var temp = {}; // to please the compiler
+// trace("// new Matrix(1, 2, 3, {})");
+// trace(new Matrix(1, 2, 3, temp));
+trace("");
+
+trace("// new Matrix(1, 2, 3, 4, 5, 6) .identity()");
+var matrix = new Matrix(1, 2, 3, 4, 5, 6);
+matrix.identity();
+trace(matrix);
+trace("");
+trace("");
+
+trace("/// Clones");
+matrix = new Matrix(1, 2, 3, 4, 5, 6);
+var cloned = matrix.clone();
+trace("// matrix");
+trace(matrix);
+trace("");
+
+trace("// cloned");
+trace(cloned);
+trace("");
+
+trace("// matrix === cloned");
+trace(matrix === cloned);
+trace("");
+trace("");
+
+// cannot use NaN values with approximate tester
+// var temp = {};
+// matrix = new Matrix(1, 2, temp, 4, 5, 6);
+// cloned = matrix.clone();
+// trace("// matrix");
+// trace(matrix);
+// trace("");
+
+trace("// cloned");
+trace(cloned);
+trace("");
+
+trace("// matrix === cloned");
+trace(matrix === cloned);
+trace("");
+trace("");
+
+
+trace("/// scale");
+trace("// matrix");
+matrix = new Matrix();
+trace(matrix);
+trace("");
+
+trace("// matrix.scale(3, 5)");
+matrix.scale(3, 5);
+trace(matrix);
+trace("");
+trace("");
+
+trace("// matrix");
+matrix = new Matrix(2, 0, 0, 2, 100, 100);
+trace(matrix);
+trace("");
+
+trace("// matrix.scale(7, 11)");
+matrix.scale(7, 11);
+trace(matrix);
+trace("");
+trace("");
+
+trace("// matrix");
+matrix = new Matrix(1, 2, 3, 4, 5, 6);
+trace(matrix);
+trace("");
+
+trace("// matrix.scale(13, 17)");
+matrix.scale(13, 17);
+trace(matrix);
+trace("");
+trace("");
+
+trace("/// rotate");
+matrix = new Matrix();
+trace(matrix);
+trace("");
+
+trace("// matrix.rotate(0)");
+matrix.rotate(0);
+trace(matrix);
+trace("");
+trace("");
+
+trace("/// rotate");
+trace("// matrix");
+matrix = new Matrix();
+trace(matrix);
+trace("");
+
+trace("// matrix.rotate(0.5)");
+matrix.rotate(0.5);
+trace(matrix);
+trace("");
+trace("");
+
+trace("// matrix");
+matrix = new Matrix(1, 2, 3, 4, 5, 6);
+trace(matrix);
+trace("");
+
+trace("// matrix.rotate(0)");
+matrix.rotate(0);
+trace(matrix);
+trace("");
+trace("");
+
+trace("// matrix");
+matrix = new Matrix(1, 2, 3, 4, 5, 6);
+trace(matrix);
+trace("");
+
+trace("// matrix.rotate((90/180)*Math.PI)");
+matrix.rotate((90/180)*Math.PI);
+trace(matrix);
+trace("");
+trace("");
+
+trace("/// translate");
+trace("// matrix");
+matrix = new Matrix();
+
+trace("// matrix.translate(3, 5)");
+matrix.translate(3, 5);
+trace(matrix);
+trace("");
+trace("");
+
+trace("// matrix");
+matrix = new Matrix(2, 0, 0, 2, 100, 100);
+
+trace("// matrix.translate(7, 11)");
+matrix.translate(7, 11);
+trace(matrix);
+trace("");
+trace("");
+
+trace("/// concat");
```

Comment:
The traced output is not really helpful here in itself. Also, maybe rotation matrices could also be added here for completeness. And all kinds of matrices on both the left and right side (as this and as parameter) as well, perhaps.

Reply:
Added comments for each step. Not sure what do you mean by right side test. Concat only support `this=this*other` operation.

Reply:
Yes, that's why I tried to clarify. I meant:
`rotationmatrix.concat(translationmatrix)`
and
`translationmatrix.concat(rotationmatrix)`
And so on for all combinations.

Reply:
Implemented for all posible combinations of basic transform matrices and application of all combinations of basic transform matrices on other matrix in all posible orders